src/main/java/ca/corefacility/bioinformatics/irida/ria/web/projects/metadata/ProjectSampleMetadataAjaxController.java

```diff
@@ -64,26 +64,27 @@ public ProjectSampleMetadataAjaxController(UIMetadataImportService metadataImpor
 	@PostMapping("/upload/setSampleColumn")
 	@ResponseBody
 	public ResponseEntity<AjaxResponse> setProjectSampleMetadataSampleId(HttpSession session,
-			@RequestParam long projectId, @RequestParam String sampleNameColumn) {
+			@RequestParam Long projectId, @RequestParam String sampleNameColumn) {
 		return ResponseEntity.ok(new AjaxSuccessResponse(
 				metadataImportService.setProjectSampleMetadataSampleId(session, projectId, sampleNameColumn)));
 	}
 
 	/**
 	 * Save uploaded metadata to the
 	 *
-	 * @param locale    {@link Locale} of the current user.
-	 * @param session   {@link HttpSession}
-	 * @param projectId {@link Long} identifier for the current project
+	 * @param locale      {@link Locale} of the current user.
+	 * @param session     {@link HttpSession}
+	 * @param projectId   {@link Long} identifier for the current project
+	 * @param sampleNames {@link List} of {@link String} sample names
 	 * @return {@link Map} of potential errors.
 	 */
 	@PostMapping("/upload/save")
 	@ResponseBody
 	public ResponseEntity<ProjectSampleMetadataResponse> saveProjectSampleMetadata(Locale locale, HttpSession session,
-			@RequestParam long projectId) {
+			@RequestParam Long projectId, @RequestParam List<String> sampleNames) {
 
 		ProjectSampleMetadataResponse response = metadataImportService.saveProjectSampleMetadata(locale, session,
-				projectId);
+				projectId, sampleNames);
 
 		if (response.getMessageKey()
 				.equals("success")) {
```

Comment:
You should probably throw an error instead of using this message key (again, probably something I did not a long time ago)

Reply:
Please see ef1206214a68c2b70ee0839752f0014dc5eab6ef.